src/test/java/com/mongodb/jdbc/MongoSQLResultSetTest.java

```diff
@@ -285,6 +320,186 @@ void testGetStringValues() throws Exception {
+        assertEquals("21.2", mongoSQLResultSetAllTypes.getString(ALL_DECIMAL_COL_LABEL));
+
+        // Null types
+        assertNull(mongoSQLResultSetAllTypes.getString(ALL_UNDEFINED_COL_LABEL));
+        assertNull(mongoSQLResultSetAllTypes.getString(ALL_NULL_COL_LABEL));
+
+        // Invalid types
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_BINARY_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_REGEX_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_DB_POINTER_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_SYMBOL_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_JAVASCRIPT_WITH_SCOPE_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_TIMESTAMP_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_MIN_KEY_COL_LABEL);
+                });
+        assertThrows(
+                SQLException.class,
+                () -> {
+                    mongoSQLResultSetAllTypes.getString(ALL_MAX_KEY_COL_LABEL);
+                });
+    }
+
+    @Test
+    public void testGetObjectToStringAllTypes() throws Exception {
+        // Test getObject().toString() result for each BSON type
+        assertEquals("1.0", mongoSQLResultSetAllTypes.getObject(ALL_DOUBLE_COL_LABEL).toString());
+        assertEquals("str", mongoSQLResultSetAllTypes.getObject(ALL_STRING_COL_LABEL).toString());
+        assertEquals(
+                "{\"x\": 10, \"y\": {\"$oid\": \"" + ALL_OBJECT_ID_VAL.toString() + "\"}}",
+                mongoSQLResultSetAllTypes.getObject(ALL_DOC_COL_LABEL).toString());
+        assertEquals(
+                "[7, 8, 9]", mongoSQLResultSetAllTypes.getObject(ALL_ARRAY_COL_LABEL).toString());
+        // TODO: Java driver doesn't seem to serialize binary using extended json
+        //        assertEquals("{\"$binary\": {\"base64\": \"\", \"subType\": \"0\"}}", mongoSQLResultSetAllTypes.getObject(ALL_BINARY_COL_LABEL).toString());
```

Comment:
Still need to look into why this is happening. Even using the java driver's extended json serialization methods results in `BsonBinary` being stringified in the wrong format:
```
Expected : {"$binary": {"base64": "", "subType": "0"}}
Actual   : BsonBinary{type=0, data=[]}
```
Could be a misunderstanding of the java driver api, or could be a bug in the java driver. My current guess is it is the latter case, since the binary codepath is the same as all the types.OTHER bson types' codepath and they all get stringified as ext json.

Reply:
Oh I see. Obvious mistake. The `getObject()` method returns a `BsonBinary` for binary types because bson binary maps to `Types.BINARY`, not `Types.OTHER`. Easy fix.